src/main/java/org/llorllale/cactoos/matchers/Mismatches.java

```diff
@@ -0,0 +1,156 @@
+ *      new Mismatches<>(
+ *          String.format("Tom%nJohn%n"),
+ *          "Lines are <[Tom, Mike]>",
+ *          "<[Tom, John]>"
+ *      )
+ *  ).affirm();
+ *  }</pre>
+ *
+ * @param <X> Type of item.
+ * @since 1.0.0
+ * @todo #106:30min Convert all the Matcher tests to use Mismatches
+ *  instead of checking directly the output of mismatch methods or
+ *  the exception of Assertion.
+ * @todo #106:30min Add extra tests for this class to validate all
+ *  the different constructors and ensure they are coherent with how
+ *  Assertion is working and throwing errors.
+ * @checkstyle ProtectedMethodInFinalClassCheck (200 lines)
+ */
+public final class Mismatches<X> extends TypeSafeDiagnosingMatcher<Matcher<X>> {
+
+    /**
+     * The testing arguments for the target matcher.
+     */
+    private final X args;
+
+    /**
+     * The expected mismatch message.
+     */
+    private final Text message;
+
+    /**
+     * Ctor.
+     * @param args The testing arguments for the matcher.
+     * @param expected The expected portion of the mismatch message.
+     * @param actual The actual portion of the mismatch message.
+     */
+    public Mismatches(
+        final X args,
+        final String expected,
+        final String actual
+    ) {
+        this(args, new TextOf(expected), new TextOf(actual));
+    }
+
+    /**
+     * Ctor.
+     * @param args The testing arguments for the matcher.
+     * @param expected The expected portion of the mismatch message.
+     * @param actual The actual portion of the mismatch message.
+     */
+    public Mismatches(
+        final X args,
+        final Text expected,
+        final Text actual
+    ) {
+        this(
+            args,
+            new Joined(
+                new TextOf("\n"),
+                new TextOf(""),
+                new Joined(new TextOf(""), new TextOf("Expected: "), expected),
+                new Joined(new TextOf(""), new TextOf(" but was: "), actual)
+            )
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param args The testing arguments for the matcher.
+     * @param message The expected mismatch message.
+     */
+    public Mismatches(final X args, final String message) {
+        this(args, new TextOf(message));
+    }
+
+    /**
+     * Ctor.
+     * @param args The testing arguments for the matcher.
+     * @param message The expected mismatch message.
+     */
+    public Mismatches(final X args, final Text message) {
+        super();
+        this.args = args;
+        this.message = message;
+    }
+
+    @Override
+    public void describeTo(final Description desc) {
+        desc.appendText("Mismatches ")
+            .appendValue(this.args)
+            .appendText(" with message ")
+            .appendValue(this.message);
+    }
+
+    // @todo #106:30min Add a description in case the mismatch fails.
+    //  And then introduce some tests to validate that the description
+    //  is properly constructed.
```

Comment:
@victornoel just a (totally minor) style concern: I always see puzzles with `/* */` instead of `//`. Although this is valid, I think that sticking to the same formatting everywhere can be better for the project.

Reply:
@scristalli good point, but if I use `/* */` I get the following error from qulice:
```
Checkstyle: src/main/java/org/llorllale/cactoos/matchers/Mismatches.java[145]: Overridden methods should not have Javadoc (NoJavadocForOverriddenMethodsCheck)
```